src/Relude/Extra/Foldable1.hs

```diff
@@ -1,54 +1,116 @@
+{-# LANGUAGE InstanceSigs #-}
+
 module Relude.Extra.Foldable1
     ( Foldable1 (..)
-    , traverse1_
-    , for1_
     ) where
 
-import Relude.Applicative (Applicative, pass)
-import Relude.Base (Ord)
-import Relude.Foldable.Reexport (Foldable (foldr))
-import Relude.Function (flip, id, (.))
-import Relude.Functor.Reexport (Compose (..), Identity (..))
-import Relude.Monoid (Semigroup, (<>))
+import Relude hiding (Product (..), Sum (..))
+
+import Data.Functor.Product (Product (..))
+import Data.Functor.Sum (Sum (..))
+import qualified Data.Semigroup as SG
 
-import Control.Applicative ((*>))
-import Data.List.NonEmpty (NonEmpty (..))
-import Data.Semigroup (First (..), Last (..), Max (..), Min (..))
+-- | The class of foldable data structures that cannot be empty.
 
 class Foldable t => Foldable1 t where
-    foldMap1   :: Semigroup m => (a -> m) -> t a -> m
+    {- | Map each element of the non-empty structure to a semigroup, and combine the results.
```

Comment:
Could you please add `{-# MINIMAL #-}` pragma to the top of this type class? Like how it's done for `Foldable`.

* https://hackage.haskell.org/package/base-4.11.1.0/docs/src/Data.Foldable.html#Foldable